app/models/assignment_invitation.rb

```diff
@@ -0,0 +1,57 @@
+class AssignmentInvitation < ActiveRecord::Base
+  belongs_to :assignment
+
+  validates :key, presence:   true
+  validates :key, uniqueness: true
+
+  after_initialize :assign_key
+
+  def redeem(user)
+    repo_access     = find_or_create_repo_access_by_organization(user, assignment.organization)
```

Comment:
@johndbritton I'd like to also get your opinion on these two methods, not sure if I should leave them on the model or extract them out to another service object. Feeling like I should leave them here